proof/crefine/ARM_HYP/SyscallArgs_C.thy

```diff
@@ -1220,7 +1220,10 @@ lemma getSyscallArg_ccorres_foo:
      apply (simp add: word_less_nat_alt split: if_split)
     apply (rule ccorres_add_return2)
     apply (rule ccorres_symb_exec_l)
-       apply (rule_tac P="\<lambda>s. n < unat (scast n_msgRegisters :: word32) \<and> obj_at' (\<lambda>tcb. atcbContextGet (tcbArch tcb) (ARM_HYP_H.msgRegisters!n) = x!n) (ksCurThread s) s"
+       apply (rule_tac P="\<lambda>s. n < unat (scast n_msgRegisters :: word32)
```

Comment:
Should this one be `word32` or `machine_word`?

Reply:
I have no idea. It's casting `n_msgRegisters` which is an `int` due to being an enumeration, so that's `32 signed word` ... then it's trying to do a `unat`, so it's casting the `signed` away with `scast`, so technically that's to `32 word` or whatever we call it on other arches (`int_len word`?), which is 32 even on 64 bit arches? Then again `machine_word` is a suitably large word to carry it, so it seems arbitrary in this case. Since you pointed it out, what do you think? :)

Reply:
I was mostly checking if you had thought about it. If it's not really tied to machine_word it's fine to leave as is. We'll get to think properly about this bit when we attempt splitting in Refine

Reply:
ok, will leave it for now